tests/components/yessssms/test_notify.py

```diff
@@ -1,7 +1,144 @@
 """The tests for the notify yessssms platform."""
+# pylint: disable=redefined-outer-name
 import unittest
+from unittest.mock import MagicMock
 import requests_mock
+import pytest
+
+from homeassistant.setup import async_setup_component
 import homeassistant.components.yessssms.notify as yessssms
+from homeassistant.components.yessssms.const import CONF_PROVIDER
+
+from homeassistant.const import CONF_PASSWORD, CONF_RECIPIENT, CONF_USERNAME
+
+
+@pytest.fixture
+def config_data():
+    """Set valid config data."""
+    config = {
+        "notify": {
+            "platform": "yessssms",
+            "name": "sms",
+            CONF_USERNAME: "06641234567",
+            CONF_PASSWORD: "secretPassword",
+            CONF_RECIPIENT: "06509876543",
+            CONF_PROVIDER: "educom",
+        }
+    }
+    return config
+
+
+@pytest.fixture
+def valid_settings(hass, config_data):
+    """Initialize component with valid settings."""
+    yield async_setup_component(hass, "notify", config_data)
+
+
+@pytest.fixture
+def invalid_provider_settings(hass, config_data):
+    """Set invalid provider data and initalize component."""
+    config = config_data
+    config["notify"][CONF_PROVIDER] = "FantasyMobile"  # invalid provider
+    yield async_setup_component(hass, "notify", config)
+
+
+@pytest.fixture
+def invalid_login_data():
+    """Mock invalid login data."""
+    yessssms.YesssSMS.login_data_valid = MagicMock(return_value=False)
```

Comment:
We should patch the `login_data_method` instead of replacing it. Patching it is temporary during the context manager scope.

```py
path = (
    "homeassistant.components.yessssms."
    "notify.YesssSMS.login_data_valid"
)

with patch(path, return_value=False):
    yield

Reply:
ok. but the Mock wouldn't live longer would it?
every fixture get's a new context of hass, doesn't it?

Reply:
Yes, but we shouldn't rely on that. Patching is the standard way of changing a module under test.